tests/posix/common/src/nanosleep.c

```diff
@@ -143,7 +167,18 @@ static void common_lower_bound_check(int selection, clockid_t clock_id, int flag
 	zassert_equal(rem.tv_sec, 0, "actual: %d expected: %d", rem.tv_sec, 0);
 	zassert_equal(rem.tv_nsec, 0, "actual: %d expected: %d", rem.tv_nsec, 0);
 
-	actual_ns = k_cyc_to_ns_ceil64((now - then * selection));
+	switch (selection) {
+	case SELECT_NANOSLEEP:
+		/* exp_ns and actual_ns are relative (i.e. durations) */
+		actual_ns = k_cyc_to_ns_ceil64(now + then);
+		break;
+	case SELECT_CLOCK_NANOSLEEP:
+		/* exp_ns and actual_ns are absolute (i.e. time-points) */
+		actual_ns = k_cyc_to_ns_ceil64(now);
+		break;
+	default:
```

Comment:
Please add an assertion here to fail in the default case.

Reply:
I see, annd add a unreachable assert in default 
'''
zassert_unreachable();
'''